eng/common/TestResources/build-test-resource-config.yml

```diff
@@ -8,14 +8,15 @@ parameters:
   # EnvVars is used to help diagnose variable conflict issues early
   - name: EnvVars
     type: object
-    default: null
+    default: {}
   - name: SubscriptionConfigurationFilePaths
     type: object
     default: null
 
 steps:
   - task: AzurePowerShell@5
     displayName: Set Pipeline Subnet Info
+    condition: ne(variables['Pool'], '')
```

Comment:
Should also add succeeded into the condition otherwise this might try to run when earlier steps have failed.

Reply:
https://github.com/Azure/azure-sdk-tools/pull/8572